### PR TITLE
Parallel Algorithms for scan partitioner for small number of elements

### PR DESCRIPTION
**Pull request by dcbdan**
Works for a small number of elements. Also created a regression test for #1390.

### CODE REVIEW

**Comment by hkaiser** [comment]
This fixes #1389: Parallel algorithms relying on scan partitioner break for small number of elements . @dcbdan: does this also fix #1390, or does it just add a regression test for it?

File: tests/regressions/parallel/scan_different_inits.cpp [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+
+    HPX_TEST(std::equal(b.begin(), b.end(), b_ans.begin()));
+    HPX_TEST(std::equal(c.begin(), c.end(), c_ans.begin()));
+    HPX_TEST(std::equal(d.begin(), d.end(), d_ans.begin()));
+    HPX_TEST(std::equal(e.begin(), e.end(), e_ans.begin()));
+}
+
+int hpx_main(boost::program_options::variables_map& vm)
+{
+    std::vector<int> a1{1,1,2,2,3,3,4,4};
```
Comment (hkaiser) [inline tests/regressions/parallel/scan_different_inits.cpp]: I'm not sure if all compilers supported by us support initializer lists. @K-ballo?

**Comment by dcbdan** [comment]
@hkaiser: Just adds regression tests for #1390

File: tests/regressions/parallel/scan_different_inits.cpp (hunk 1 above)
Comment (K-ballo) [inline tests/regressions/parallel/scan_different_inits.cpp]: Theoretically yes, but there are some issues with msvc and intel13. I think this simple case should be ok when used with a C++11 standard library implementation.

**Comment by hkaiser** [comment]
Ok, thanks! I'm fine with merging this.

**Pull request merged**